### ISSUE TITLE
Show visual label for Archives block in dropdown mode

### ISSUE THREAD

**Issue by tofumatt** [WordPress/gutenberg#11002]
See: 
https://github.com/WordPress/gutenberg/pull/10912#issuecomment-432602486

Right now the Archives block in dropdown mode does not show a visual label. We should _at least_ show one in editing mode, because otherwise it's a bit confusing seeing a dropdown without context in the editor:

<img width="841" alt="screenshot 2018-10-24 at 12 59 33" src="https://user-images.githubusercontent.com/90871/47428934-a9440280-d78c-11e8-884d-784c08cd8406.png">

**Comment by afercia** [comment]
Worth considering to make the label visible also in the front-end. Quoting from https://github.com/WordPress/gutenberg/issues/10871

> I'd tend to think the labels shouldn't be visually hidden. It's not up to Gutenberg to determine if the labels should be visible or not in the front-end. I'd propose to consider to remove the screen-reader-text CSS class in the admin and in the front-end, as their visibility should be up to the theme.

> Noting it's a different case from what core does with the related current widgets, as these widgets have an option to display a title. In Gutenberg instead, there's no title. Making the labels visible would address this issue and allow themes to style them in the way they want.

Screenshot of the Archives and Categories selects in the front-end:

<img width="627" alt="screen shot 2018-10-21 at 16 52 33" src="https://user-images.githubusercontent.com/1682452/47268513-57179d00-d552-11e8-9995-2dfc997f13b1.png">

**Comment by youknowriad** [comment]
Moving out of the RC, these are new blocks and already have a label. So not really critical.
